examples/random/Dockerfile

```diff
@@ -0,0 +1,20 @@
+# This Dockerfile builds an image for a client_golang example.
+#
+# Use as (from the root for the client_golang repository):
+#    docker build -f examples/$name/Dockerfile -t prometheus/golang-example-$name .
+
+# Builder image, where we build the example.
+FROM golang:1.9.0 AS builder
```

Comment:
Maybe I miss something but wouldn't this be simpler?
```
FROM golang:1.9.0 AS builder
COPY . /go/src/github.com/prometheus/client_golang
WORKDIR /go/src/github.com/prometheus/client_golang/examples/random
RUN CGO_ENABLED=0 GOOS=linux go build -a -tags netgo -ldflags '-w'
```

Reply:
That was to share the `go get -d` layer between images.

Reply:
Ah I see what you mean.. Well, guess then it's fine. I personally would prefer a simpler, less optimized Dockerfile but really shouldn't matter.

Reply:
So overall LGTM and leaving the final decision over that to Björn